src/e2e/global-teardown.ts

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import https from "https";
+import { readFileSync, writeFileSync } from "fs";
+import { CustomTestResult } from "../../E2EReporter";
+
+async function globalTearDown() {
+  // check env for cron, teardown is only needed for teardown run
+  if (process.env.IS_CRON) {
+    // get data from e2e-reports branch files -- these are api calls
+    const {
+      existingFailedReports,
+      existingFlakeReports,
+      existingFullReports,
+      newFullReportJson,
+      newFailReportJson,
+      newFlakeReportJson,
+    } = await getExistingAndNewReports();
+
+    // merge new and existing reports
+    existingFullReports.push(newFullReportJson as unknown as CustomTestResult);
+    if (newFlakeReportJson.length)
+      existingFlakeReports.push(
+        newFlakeReportJson as unknown as CustomTestResult,
+      );
+    if (newFailReportJson.length)
+      existingFailedReports.push(
+        newFailReportJson as unknown as CustomTestResult,
+      );
```

Comment:
Interesting that ESLint didn't trigger something about [`curly`](https://eslint.org/docs/latest/rules/curly) rule, but maybe we don't enforce curly braces for single statements. 🤷